webdictionary/src/main/java/com/eis0/kademlianetwork/ConnectionHandler.java

```diff
@@ -0,0 +1,79 @@
+package com.eis0.kademlianetwork;
+
+import com.eis.smslibrary.SMSManager;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis0.kademlia.SMSKademliaNode;
+
+/**
+ * Class handling the first time connection to a kademlia network.
+ * Gives methods to join someone's network or to invite someone to my
+ * network
+ *
+ * @author Marco Cognolato
+ */
+public class ConnectionHandler {
+
+    /**
+     * Invites a given valid SMSPeer to join my kademlia network
+     *
+     * @param peer The peer to invite to my network
+     */
+    public void inviteToJoin(SMSPeer peer) {
+        String messageRequest = RequestTypes.JoinPermission.ordinal() + "";
+        SMSMessage message = new SMSMessage(peer, messageRequest);
+        SMSManager.getInstance().sendMessage(message);
+    }
+
+    /**
+     * Asks a given valid peer if I can join HIS Kademlia Network
+     *
+     * @param peer The peer to ask to
+     */
+    public void askToJoin(SMSPeer peer) {
+        //asking to join has the same functionality as inviting someone to join
+        inviteToJoin(peer);
+    }
+
+    /**
+     * Accepts the request sent by an SMSPeer (by notifying him)
+     *
+     * @param peer The peer who sent the request (and to notify back)
+     */
+    public static void sendAcceptRequest(SMSPeer peer) {
+        /*
+         * If both the other peer sent me a request to join his network, or
+         * a request to invite me to his network, I have to update my routing table
+         * from scratch adding him to the table as a contact, then I have to send
+         * an acceptJoin event, so he can do the same thing.
+         * This serves to more strongly fuse both networks by
+         * creating contacts from the other network
+         * */
+
+        String messageRequest = RequestTypes.AcceptJoin.ordinal() + "";
+        SMSMessage message = new SMSMessage(peer, messageRequest);
+        SMSManager.getInstance().sendMessage(message);
+
+        acceptRequest(peer);
+    }
+
+    /**
+     * Accepts the request sent by an SMSPeer
+     *
+     * @param peer The peer who sent the request
+     */
+    public static void acceptRequest(SMSPeer peer) {
+        /*
+         * If both the other peer sent me a request to join his network, or
+         * a request to invite me to his network, I have to update my routing table
+         * from scratch adding him to the table as a contact
```

Comment:
Isn't this part of the comment redundant? I mean... It looks like it's copy-paste of the one above...
I think you could optimize it by putting acceptRequest method before sendAcceptRequest, so you'd be able to avoid lines of redundant comments...

Reply:
Yes, the comment is copy pasted, though there's a slightly difference which maybe I should point out more clearly.
there's this piece of comment in `sendAcceptRequest` which is not in `acceptRequest`:
> then I have to send an acceptJoin event, so he can do the same thing.

The reason most of the comment is the same is to have every functions self documented, so when I look at how a function works I can stay inside that function, instead of having to look around for the documentation.

I'll change a part of the comment to make that difference pop up more easily, but I think I'll still keep these functions in this order since `sendAcceptRequest` uses `acceptRequest`, so having them in order is (at least in my opinion) better to read

Reply:
Alright, updated. Is it better now?

Reply:
Not really... It looks kinda confusing to me... :(
It looks like:
[sendAcceptRequest] When I receive a request to join his network, or (when I receive) a request to join his network I have to send an acceptJoin event, to notify him and let him do his work.
[acceptRequest]  If both the other peer sent me a request to join his network, or (the other peer sent me) a request to invite me to his network, I have to update my routing table from scratch adding him to the table as a contact

Did you mean to say these things above ⬆️ , or these things below ⬇️ ?

[sendAcceptRequest] When I receive a request to join his network, or when I request to join his network I have to send an acceptJoin event, to notify him and let him do his work.
[acceptRequest]  If both the other peer sent me a request to join his network, or I request the other peer to join his network, I have to update my routing table from scratch adding him to the table as a contact

Reply:
What happens is:
either,
- I send a request to join his network, and he sends back an AcceptJoin message
- Or I send an invite for him to join my network, and he sends back and AcceptJoin message

by "AcceptJoin" I intend both the "I just got invited to another network and I'm accepting" and the "Someone asked to enter my network and I'm accepting"

Reply:
That's because when you look at what happens in both cases, you see that they do the same thing:
both add the other person to their own routing table and then update it

Reply:
Oh ok now I think I get it... Thank you! :)

Reply:
So? Do you think I should write it better now that you understand?

Reply:
Personally, I think you could actually write it better...
There can be people which are more retarded than me in this world...